filelog.go

```diff
@@ -316,13 +411,18 @@ func (w *FileLogWriter) handleRotate(rotateTime time.Time) error {
 			w.closeLogFile()
 
 			// Rename the file to its newfound home
-			err = os.Rename(w.filename, fname)
+			err = os.Rename(w.filename, rotatedName)
 			if err != nil {
 				return fmt.Errorf("Rotate: %s\n", err)
 			}
 		}
 	}
 
+	// If we're configured to archive files, signal the background goroutine
+	if w.filesToKeep > 0 {
```

Comment:
I don't know how I missed it before - shouldn't this be inside the `if w.rotate` block? Otherwise we'll just send an empty string on this channel.